### PR TITLE
Loading Gif when submitting student/teacher form

### PR DESCRIPTION
**Pull request by echang9**

### CODE REVIEW

File: app/assets/javascripts/components/form/student_form.es6.jsx [hunk 1]
```diff
@@ -240,6 +241,10 @@ class StudentForm extends React.Component {
     this.setState({ showWaiverModal: false });
   }
 
+  stopLoading() {
+    this.setState( {loading : false});
```
Comment (eric-ls) [inline app/assets/javascripts/components/form/student_form.es6.jsx]: nit: spacing

File: app/assets/javascripts/components/form/student_form.es6.jsx [hunk 2]
```diff
@@ -248,7 +253,7 @@ class StudentForm extends React.Component {
     for (var i = 0; i < eventArray.length; i++) {
       availabilityArray = availabilityArray.concat(range_to_array(eventArray[i]['start'], eventArray[i]['end']));
     }
-    this.setState({ availability: availabilityArray });
+    this.setState({ availability: availabilityArray, loading: true });
```
Comment (eric-ls) [inline app/assets/javascripts/components/form/student_form.es6.jsx]: if theres more than 1 key being set I think it should be multiline

File: app/assets/javascripts/components/form/student_form.es6.jsx [hunk 3]
```diff
@@ -466,10 +473,13 @@ class StudentForm extends React.Component {
 
   createStudent(customer) {
     var reject = (response) => {
-      this.setState({ errors: response.errors });
+      this.setState({ errors: response.errors, loading: false });
```
Comment (eric-ls) [inline app/assets/javascripts/components/form/student_form.es6.jsx]: same as above, multiline

File: app/assets/javascripts/components/form/teacher_form.es6.jsx [hunk 4]
```diff
@@ -254,6 +255,10 @@ class TeacherForm extends React.Component {
     this.setState({ showWaiverModal: false });
   }
 
+  stopLoading() {
+    this.setState( {loading : false});
```
Comment (eric-ls) [inline app/assets/javascripts/components/form/teacher_form.es6.jsx]: nit: spacing

File: app/assets/javascripts/components/form/teacher_form.es6.jsx [hunk 5]
```diff
@@ -262,7 +267,7 @@ class TeacherForm extends React.Component {
     for (var i = 0; i < eventArray.length; i++) {
       availabilityArray = availabilityArray.concat(range_to_array(eventArray[i]['start'], eventArray[i]['end']));
     }
-    this.setState({ availability: availabilityArray })
+    this.setState({ availability: availabilityArray, loading: true })
```
Comment (eric-ls) [inline app/assets/javascripts/components/form/teacher_form.es6.jsx]: same, multiline

File: app/assets/javascripts/components/form/teacher_form.es6.jsx [hunk 6]
```diff
@@ -363,9 +385,18 @@ class TeacherForm extends React.Component {
...
           account_holder_type: stripe_account_holder_type
         }, this.stripeResponseHandler.bind(this));
+      } else {
+        this.stopLoading();
       }
     } else {
-      this.createTeacher();
+      instrument_errors = await this.validateInstruments();
+      if (!(Object.keys(instrument_errors).length === 0)) {
+        console.log("wtf");
```
Comment (eric-ls) [inline app/assets/javascripts/components/form/teacher_form.es6.jsx]: remove log lol

File: app/assets/javascripts/components/form/teacher_form.es6.jsx [hunk 7]
```diff
@@ -521,31 +555,16 @@ class TeacherForm extends React.Component {
   createTeacher(accountResponse) {
     const { teach_for_free } = this.state;
     var reject = (response) => {
-      this.setState({ errors: response.errors });
+      this.setState({ errors: response.errors, loading: false });
```
Comment (eric-ls) [inline app/assets/javascripts/components/form/teacher_form.es6.jsx]: same?

**Comment by eric-ls** [comment]
@echang9 can you also include a screenshot?

**Comment by echang9** [comment]
Here's the screenshot! Thanks @eric-ls 

screen shot 2017-03-16 at 2 18 52 pm

**Pull request merged**